### ISSUE TITLE
Upgrading a Docker alpine setup from 1.1.0 to 1.2.0 yields Internal Server Error

### ISSUE THREAD

**Issue by almereyda** [hackmdio/codimd#876]
When upgrading from 1.1.0 to 1.2.0 we receive `Internal Server Error` on the frontend.
The service reports:

```
# docker-compose pull web && docker-compose stop web && docker-compose rm -f web && docker-compose up -d web
Pulling web (hackmdio/hackmd:alpine)...                                                                      
alpine: Pulling from hackmdio/hackmd                                           
Digest: sha256:73d3fedf33bd135fb4cc2abc5cbf704b58aab60131a78632eb557194a06a0e54
Status: Image is up to date for hackmdio/hackmd:alpine                                    
Stopping hackallmendeio_web_1 ... done                                                   
Going to remove hackallmendeio_web_1                                                           
Removing hackallmendeio_web_1 ... done                                                                                                                                                                                                                
hackallmendeio_postgres_1 is up-to-date                                                   
Creating hackallmendeio_web_1                                                              
# docker-compose logs -f web                                       
Attaching to hackallmendeio_web_1                                                                                           
web_1       | 2018/06/30 20:11:16 Waiting for: tcp://postgres:5432                                                      
web_1       | 2018/06/30 20:11:16 Connected to tcp://postgres:5432                                                                      
web_1       |                                                                                                  
web_1       | Sequelize [Node: 8.11.3, CLI: 2.8.0, ORM: 3.30.4]                                                                         
web_1       |                                                                                                
web_1       | Parsed url postgres://hackmd:*****@postgres:5432/hackmd                                      
web_1       | (node:28) DeprecationWarning: Using the automatically created return value from client.query as an event emitter is deprecated and will be removed in pg@7.0. Please see the upgrade guide at https://node-postgres.com/guides/upgrading
web_1       | == 20150702001020-update-to-0_3_1: migrating =======                                                          
web_1       | { SequelizeDatabaseError: column "shortid" of relation "Notes" already exists    
web_1       |     at Query.formatError (/hackmd/node_modules/sequelize/lib/dialects/postgres/query.js:357:14)               
web_1       |     at Result.<anonymous> (/hackmd/node_modules/sequelize/lib/dialects/postgres/query.js:88:19)
web_1       |     at emitOne (events.js:116:13)                                                
web_1       |     at Result.emit (events.js:211:7)                                        
web_1       |     at Result.Query.handleError (/hackmd/node_modules/pg/lib/query.js:163:8)     
web_1       |     at Client.<anonymous> (/hackmd/node_modules/pg/lib/client.js:188:26)    
web_1       |     at emitOne (events.js:116:13)           
web_1       |     at Connection.emit (events.js:211:7)                       
web_1       |     at Socket.<anonymous> (/hackmd/node_modules/pg/lib/connection.js:133:12)
web_1       |     at emitOne (events.js:116:13)                                                
web_1       |     at Socket.emit (events.js:211:7)                                        
web_1       |     at addChunk (_stream_readable.js:263:12)                                                                  
web_1       |     at readableAddChunk (_stream_readable.js:250:11)
web_1       |     at Socket.Readable.push (_stream_readable.js:208:10)       
web_1       |     at TCP.onread (net.js:597:20)                                          
web_1       |   name: 'SequelizeDatabaseError',                                                
web_1       |   message: 'column "shortid" of relation "Notes" already exists',           
web_1       |   parent:                                                                                                     
web_1       |    { error: column "shortid" of relation "Notes" already exists
web_1       |     at Connection.parseE (/hackmd/node_modules/pg/lib/connection.js:567:11)
web_1       |     at Connection.parseMessage (/hackmd/node_modules/pg/lib/connection.js:391:17)
web_1       |     at Socket.<anonymous> (/hackmd/node_modules/pg/lib/connection.js:129:22)     
web_1       |     at emitOne (events.js:116:13)                                           
web_1       |     at Socket.emit (events.js:211:7)
web_1       |     at addChunk (_stream_readable.js:263:12)    
web_1       |     at readableAddChunk (_stream_readable.js:250:11)                                                          
web_1       |     at Socket.Readable.push (_stream_readable.js:208:10)
web_1       |     at TCP.onread (net.js:597:20)                              
web_1       |      name: 'error',                                                                                           
web_1       |      length: 131,                                                                                         
web_1       |      severity: 'ERROR',                                                                                                   
web_1       |      code: '42701',                                                                                       
web_1       |      detail: undefined,                                                                                                                                                                                                                  
web_1       |      hint: undefined,                                                                                                                                                                                                                  
web_1       |      position: undefined,                                                                    
web_1       |      internalPosition: undefined,                                                                             
web_1       |      internalQuery: undefined,                                     
web_1       |      where: undefined,                                         
web_1       |      schema: undefined,                                                    
web_1       |      table: undefined,                                                           
web_1       |      column: undefined,                                                     
web_1       |      dataType: undefined,                                                                                     
web_1       |      constraint: undefined,                                                                               
web_1       |      file: 'tablecmds.c',                                                                                                                                                                                                                
web_1       |      line: '5110',                                                                                                                                                                                                                     
web_1       |      routine: 'check_for_column_name_collision',                                             
web_1       |      sql: 'ALTER TABLE "public"."Notes" ADD COLUMN "shortid" VARCHAR(255) NOT NULL DEFAULT \'0000000000\';' },
web_1       |   original:                                                                                                   
web_1       |    { error: column "shortid" of relation "Notes" already exists                                           
web_1       |     at Connection.parseE (/hackmd/node_modules/pg/lib/connection.js:567:11)                                                                                                                                                              
web_1       |     at Connection.parseMessage (/hackmd/node_modules/pg/lib/connection.js:391:17)                                                                                                                                                      
web_1       |     at Socket.<anonymous> (/hackmd/node_modules/pg/lib/connection.js:129:22)                 
web_1       |     at emitOne (events.js:116:13)                                
web_1       |     at Socket.emit (events.js:211:7)                               
web_1       |     at addChunk (_stream_readable.js:263:12)
web_1       |     at readableAddChunk (_stream_readable.js:250:11)
web_1       |     at Socket.Readable.push (_stream_readable.js:208:10)
web_1       |     at TCP.onread (net.js:597:20)
web_1       |      name: 'error',   
web_1       |      length: 131,      
web_1       |      severity: 'ERROR',  
web_1       |      code: '42701',        
web_1       |      detail: undefined,  
web_1       |      hint: undefined,
web_1       |      position: undefined,                       
web_1       |      internalPosition: undefined,                                                                             
web_1       |      internalQuery: undefined,                                                                            
web_1       |      where: undefined,                                                                                                                                                                                                                   
web_1       |      schema: undefined,                                                                                                                                                                                                                
web_1       |      table: undefined,                                                                       
web_1       |      column: undefined,                                          
web_1       |      dataType: undefined,                                          
web_1       |      constraint: undefined,
web_1       |      file: 'tablecmds.c',
web_1       |      line: '5110',
web_1       |      routine: 'check_for_column_name_collision',
web_1       |      sql: 'ALTER TABLE "public"."Notes" ADD COLUMN "shortid" VARCHAR(255) NOT NULL DEFAULT \'0000000000\';' },
web_1       |   sql: 'ALTER TABLE "public"."Notes" ADD COLUMN "shortid" VARCHAR(255) NOT NULL DEFAULT \'0000000000\';' }
web_1       | 2018-06-30T20:11:21.877Z - warn: Using legacy HMD prefix for environment variables. Please change your variables in future. For details see: https://github.com/hackmdio/codimd#environment-variables-will-overwrite-other-server-configs
web_1       | (node:1) DeprecationWarning: Using the automatically created return value from client.query as an event emitter is deprecated and will be removed in pg@7.0. Please see the upgrade guide at https://node-postgres.com/guides/upgrading
web_1       | >> WARNING: PostgreSQL does not support TEXT with options. Plain `TEXT` will be used instead.
web_1       | >> Check: http://www.postgresql.org/docs/9.4/static/datatype.html
web_1       | 2018-06-30T20:11:23.053Z - info: HTTP Server listening at port 3000
```

**Comment by almereyda** [comment]
Why is it even trying to run a migration to 0.3.1?

**Comment by ccoenen** [comment]
those were added in after-the-fact about a year ago, if I remember correctly: https://github.com/hackmdio/codimd/pull/406

**Comment by hcaloto** [comment]
I faced the same issue when upgrading from 1.1.1. Looking at the pull request #877 I've solved the problem by modifying all migrating files so that duplicated keys are ignored. 

Here is the patch I've used as a workaround, just in case it can help someone with the same issue:

```
diff --git a/lib/migrations/20150702001020-update-to-0_3_1.js b/lib/migrations/20150702001020-update-to-0_3_1.js
index 40d9c97..e7cf398 100644
--- a/lib/migrations/20150702001020-update-to-0_3_1.js
+++ b/lib/migrations/20150702001020-update-to-0_3_1.js
@@ -20,6 +20,12 @@ module.exports = {
         type: Sequelize.INTEGER,
         defaultValue: 0
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'shortid'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160112220142-note-add-lastchange.js b/lib/migrations/20160112220142-note-add-lastchange.js
index b4e111b..6cf4843 100644
--- a/lib/migrations/20160112220142-note-add-lastchange.js
+++ b/lib/migrations/20160112220142-note-add-lastchange.js
@@ -7,6 +7,12 @@ module.exports = {
       return queryInterface.addColumn('Notes', 'lastchangeAt', {
         type: Sequelize.DATE
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'lastchangeuserId'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160420180355-note-add-alias.js b/lib/migrations/20160420180355-note-add-alias.js
index a043cd5..1d1a1c1 100644
--- a/lib/migrations/20160420180355-note-add-alias.js
+++ b/lib/migrations/20160420180355-note-add-alias.js
@@ -7,6 +7,12 @@ module.exports = {
       return queryInterface.addIndex('Notes', ['alias'], {
         indicesType: 'UNIQUE'
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'alias'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160515114000-user-add-tokens.js b/lib/migrations/20160515114000-user-add-tokens.js
index 4d5818b..3b321c8 100644
--- a/lib/migrations/20160515114000-user-add-tokens.js
+++ b/lib/migrations/20160515114000-user-add-tokens.js
@@ -3,6 +3,12 @@ module.exports = {
   up: function (queryInterface, Sequelize) {
     return queryInterface.addColumn('Users', 'accessToken', Sequelize.STRING).then(function () {
       return queryInterface.addColumn('Users', 'refreshToken', Sequelize.STRING)
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'accessToken'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160607060246-support-revision.js b/lib/migrations/20160607060246-support-revision.js
index bcab97e..7d23ab3 100644
--- a/lib/migrations/20160607060246-support-revision.js
+++ b/lib/migrations/20160607060246-support-revision.js
@@ -15,6 +15,12 @@ module.exports = {
         createdAt: Sequelize.DATE,
         updatedAt: Sequelize.DATE
       })
+    }).catch(function (error) {
diff --git a/lib/migrations/20150702001020-update-to-0_3_1.js b/lib/migrations/20150702001020-update-to-0_3_1.js
index 40d9c97..e7cf398 100644
--- a/lib/migrations/20150702001020-update-to-0_3_1.js
+++ b/lib/migrations/20150702001020-update-to-0_3_1.js
@@ -20,6 +20,12 @@ module.exports = {
         type: Sequelize.INTEGER,
         defaultValue: 0
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'shortid'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160112220142-note-add-lastchange.js b/lib/migrations/20160112220142-note-add-lastchange.js
index b4e111b..6cf4843 100644
--- a/lib/migrations/20160112220142-note-add-lastchange.js
+++ b/lib/migrations/20160112220142-note-add-lastchange.js
@@ -7,6 +7,12 @@ module.exports = {
       return queryInterface.addColumn('Notes', 'lastchangeAt', {
         type: Sequelize.DATE
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'lastchangeuserId'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160420180355-note-add-alias.js b/lib/migrations/20160420180355-note-add-alias.js
index a043cd5..1d1a1c1 100644
--- a/lib/migrations/20160420180355-note-add-alias.js
+++ b/lib/migrations/20160420180355-note-add-alias.js
@@ -7,6 +7,12 @@ module.exports = {
       return queryInterface.addIndex('Notes', ['alias'], {
         indicesType: 'UNIQUE'
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'alias'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160515114000-user-add-tokens.js b/lib/migrations/20160515114000-user-add-tokens.js
index 4d5818b..3b321c8 100644
--- a/lib/migrations/20160515114000-user-add-tokens.js
+++ b/lib/migrations/20160515114000-user-add-tokens.js
@@ -3,6 +3,12 @@ module.exports = {
   up: function (queryInterface, Sequelize) {
     return queryInterface.addColumn('Users', 'accessToken', Sequelize.STRING).then(function () {
       return queryInterface.addColumn('Users', 'refreshToken', Sequelize.STRING)
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'accessToken'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160607060246-support-revision.js b/lib/migrations/20160607060246-support-revision.js
index bcab97e..7d23ab3 100644
--- a/lib/migrations/20160607060246-support-revision.js
+++ b/lib/migrations/20160607060246-support-revision.js
@@ -15,6 +15,12 @@ module.exports = {
         createdAt: Sequelize.DATE,
         updatedAt: Sequelize.DATE
       })
+    }).catch(function (error) {
diff --git a/lib/migrations/20150702001020-update-to-0_3_1.js b/lib/migrations/20150702001020-update-to-0_3_1.js
index 40d9c97..e7cf398 100644
--- a/lib/migrations/20150702001020-update-to-0_3_1.js
+++ b/lib/migrations/20150702001020-update-to-0_3_1.js
@@ -20,6 +20,12 @@ module.exports = {
         type: Sequelize.INTEGER,
         defaultValue: 0
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'shortid'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160112220142-note-add-lastchange.js b/lib/migrations/20160112220142-note-add-lastchange.js
index b4e111b..6cf4843 100644
--- a/lib/migrations/20160112220142-note-add-lastchange.js
+++ b/lib/migrations/20160112220142-note-add-lastchange.js
@@ -7,6 +7,12 @@ module.exports = {
       return queryInterface.addColumn('Notes', 'lastchangeAt', {
         type: Sequelize.DATE
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'lastchangeuserId'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160420180355-note-add-alias.js b/lib/migrations/20160420180355-note-add-alias.js
index a043cd5..1d1a1c1 100644
--- a/lib/migrations/20160420180355-note-add-alias.js
+++ b/lib/migrations/20160420180355-note-add-alias.js
@@ -7,6 +7,12 @@ module.exports = {
       return queryInterface.addIndex('Notes', ['alias'], {
         indicesType: 'UNIQUE'
       })
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'alias'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160515114000-user-add-tokens.js b/lib/migrations/20160515114000-user-add-tokens.js
index 4d5818b..3b321c8 100644
--- a/lib/migrations/20160515114000-user-add-tokens.js
+++ b/lib/migrations/20160515114000-user-add-tokens.js
@@ -3,6 +3,12 @@ module.exports = {
   up: function (queryInterface, Sequelize) {
     return queryInterface.addColumn('Users', 'accessToken', Sequelize.STRING).then(function () {
       return queryInterface.addColumn('Users', 'refreshToken', Sequelize.STRING)
+    }).catch(function (error) {
+       if (error.message === "ER_DUP_FIELDNAME: Duplicate column name 'accessToken'") {
+           console.log('Migration has already run… ignoring.')
+       } else {
+           throw error
+       }
     })
   },

diff --git a/lib/migrations/20160607060246-support-revision.js b/lib/migrations/20160607060246-support-revision.js
```

**Comment by SISheogorath** [comment]
@hcaloto That's gold! thanks, that's actually the missing part I searched for as my dbs are already fully migrated without problem or having completely different issues (:eyes: looking at you heroku…)

Can you file these changes as PR please? either on top of mine or as a separate one, I would love to merge it!

**Comment by SISheogorath** [comment]
Should be fixed by #894
